onnxruntime/contrib_ops/cuda/collective/nccl_kernels.cc

```diff
@@ -35,21 +42,137 @@ static ncclDataType_t GetNcclDataType(onnxruntime::MLDataType type) {
   }
 }
 
+#ifndef USE_MPI
+namespace IPC {
+#define FLLOG LOGS_DEFAULT(VERBOSE)
+#define FLLOGERRNO LOGS_DEFAULT(WARNING) << "error:" << strerror(errno)
+
+int WriteOnRank0(ncclUniqueId* nccl_id, int word_size) {
+  int fd = socket(AF_INET,     /* network versus AF_LOCAL */
+                  SOCK_STREAM, /* reliable, bidirectional, arbitrary payload size */
+                  0);          /* system picks underlying protocol (TCP) */
+  if (fd < 0) {
+    FLLOGERRNO << (" create socket\n"); /* terminate */
+    return -1;
+  }
+
+  int32_t PortNumber = ParseEnvironmentVariableWithDefault<int32_t>("RANK0_PORT", 18888);
+
+  /* bind the server's local address in memory */
+  struct sockaddr_in saddr;
```

Comment:
Now usually we use `struct addrinfo`  instead of interacting with sockaddr or sockaddr_in directly.